cdap-ui/app/cdap/components/Footer/index.js

```diff
@@ -18,15 +18,21 @@ import React from 'react';
 import { Theme } from 'services/ThemeHelper';
 import If from '../If';
 import { objectQuery } from 'services/helpers';
+import NamespaceStore from 'services/NamespaceStore';
 require('./Footer.scss');
 
 export default function Footer() {
   const footerText = Theme.footerText;
   const footerUrl = Theme.footerLink;
   // 'project-id-30-characters-name1/instance-id-30-characters-name';
   const instanceMetadataId = objectQuery(window, 'CDAP_CONFIG', 'instanceMetadataId');
+  const [selectedNamespace, setSelectedNamespace] = React.useState(
+    NamespaceStore.getState().selectedNamespace
+  );
+  NamespaceStore.subscribe(() => setSelectedNamespace(NamespaceStore.getState().selectedNamespace));
```

Comment:
You'll need to wrap this around `useEffect` hook. useEffect is pretty similar to `componentDidMount`. In the callback function of the hook, you can return a closure that unsubscribe from the NamespaceStore update (similar to `componentWillUnmount`

Reply:
Done.